src/dump.c

```diff
@@ -1818,9 +1818,14 @@ static jl_value_t *jl_deserialize_value_singleton(jl_serializer_state *s, jl_val
     if (s->mode == MODE_MODULE) {
         // TODO: optimize the case where the value can easily be obtained
         // from an external module (tag == 6) as dt->instance
-        assert(loc != NULL && loc != HT_NOTFOUND);
-        arraylist_push(&flagref_list, loc);
-        arraylist_push(&flagref_list, (void*)pos);
+        assert(loc != HT_NOTFOUND);
+        // if loc == NULL, then the caller doesn't need the address of the instance.
```

Comment:
Would be more accurate to say "is unable to provide the address where the instance will be stored". The assertion seems like it may be correct in detecting that we can't currently compute the tindex for the field store that we'll attempt just after this.

Reply:
What do you mean by tindex?

Reply:
Oh, right. `tindex` is the codegen name for it. In jl_get_nth_field, it's called the type selector byte.

Reply:
The type selector is serialized separately though; we don't need to compute it.

Reply:
Oh, that's valid? Ok then, should be all set